### DISCUSSION TITLE
Rendering of a Big Model

### DISCUSSION THREAD

**Discussion by marcoma9023** [pelicanmapping/osgearth#2513] [Support Q&A]
Hi,

I need to create a 3D recostruction of a port.

For this purpose I created a 3D recostruction of a dock on which I placed some models:

- cranes
- containers 
- buildings

The 3D "port" is exported into KML and I converted the .dae model into .osgb.

```c++
kmlLowNode = osgEarth::Drivers::KML::load(osgEarth::URI(subArr->get(0).toString()), mapNode, kmlOption);
kmlLowNode->getOrCreateStateSet()->setMode(GL_DEPTH_TEST, true);
kmlLowNode->getOrCreateStateSet()->setDefine(OE_LIGHTING_DEFINE, osg::StateAttribute::OFF | osg::StateAttribute::PROTECTED);
```

The problem is that when the camera looks at the 3D model of the port, performance drops drastically.

I would like to understand if I'm doing something wrong or if you have any tips for managing these models to avoid lag.

One idea could be to split each model placed on the dock into a separate kml, but before proceeding I would like to understand if there are other solutions.

Thanks

**Comment by gwaldron** [comment]
Hard to say without seeing the models, but a couple of thoughts.

View each component (.osgb) model on its own (in osgviewer or osgearth_viewer) and see if one in particular is causing the problem. You can turn on the stats by hitting 's' a couple of times to see the statistics for that model.

You may be able to optimize your models by using the osgUtil::Optimizer class in OSG.

Though unlikely, sometimes a massive slowdown is caused by incorrect geometry in the model and there may have been errors in the conversion process.

Of course if these were (for example) CAD models, it's possible they are just too complex, and you might need to use a tool to simplify them or to create some lower LODs.

Good luck.
